code/obj/machinery/microwave.dm

```diff
@@ -202,122 +202,101 @@ obj/machinery/microwave/attackby(var/obj/item/O, var/mob/user)
 	*  Microwave Menu
 	*/
 
+
+/obj/machinery/microwave/ui_interact(mob/user, datum/tgui/ui)
+	ui = tgui_process.try_update_ui(user, src, ui)
+	if (!ui)
+		ui = new(user, src, "Microwave")
+		ui.open()
+
+/obj/machinery/microwave/ui_data(mob/user)
+	. = list(
+		"broken" = src.microwave_state > 0,
+		"operating" = src.operating,
+		"dirty" = src.dirty,
+		"eggs" = src.egg_amount,
+		"flour" = src.flour_amount,
+		"monkey_meat" = src.monkeymeat_amount,
+		"synth_meat" = src.synthmeat_amount,
+		"donk_pockets" = src.donkpocket_amount,
+		"other_meat" = src.humanmeat_amount,
+		"unusual_item" = src.extra_item
+		)
+
+/obj/machinery/microwave/ui_act(action, params)
+	. = ..()
+	if (.)
+		return
+	. = TRUE
```

Comment:
Do within specific actions that should trigger a UI update, not all of the time.